src/main/java/org/wso2/carbon/esb/connector/NatsConnectionPool.java

```diff
@@ -0,0 +1,125 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.esb.connector;
+
+import io.nats.client.Connection;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Connection pool for NATS publisher connections.
+ */
+class NatsConnectionPool {
+
+    private static Log log = LogFactory.getLog(NatsConnectionPool.class);
+    private static ArrayList<Connection> connectionPool = new ArrayList<>();
```

Comment:
Better to use `private static List<Connection> connectionPool = new ArrayList<>()`; whenever you are using collections